jsk_fetch_robot/fetcheus/fetch-interface.l

```diff
@@ -40,8 +40,11 @@
      (ros::warn ":angle-vector tm is not a number, use :angle-vector av tm args"))
    (let ((use-torso t))
      (if (and (member :use-torso args) (null (cadr (member :use-torso args)))) (setq use-torso nil))
-     (send self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso use-torso :clear-velocities t)
+     (send* self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso use-torso :clear-velocities t args)
    ))
+  (:rarm-angle-vector 
+   (av &optional (tm 3000) &rest args)
+   (send* self :angle-vector av tm :ctype :arm-controller args))
```

Comment:
Updated to the code below to solve some bugs:
```
(:angle-vector ;; this verison uses :angle-vector-sequence for sendding trajectory, this enable us to use :wait-interpolation method so we choose this for now
   (av &optional (tm 3000) &rest args)
   "Send joind angle to robot with self-collision motion planning, this method retuns immediately, so use :wait-interpolation to block until the motion stops.
- av : joint angle vector [rad]
- tm : (time to goal in [msec]) ;; currently this value is ignored
- use-torso : set t to use torso
"
   ;; for simulation mode
   (when (send self :simulation-modep)
     (return-from :angle-vector (send* self :angle-vector-raw av tm args)))
   ;;
   (when (not (numberp tm))
     (ros::warn ":angle-vector tm is not a number, use :angle-vector av tm args"))
   (let ((use-torso t)
	 (ctype (find-if #'(lambda (x) (gethash x (send self :get-val 'controller-table))) args)))
     (if (and (member :use-torso args) (null (cadr (member :use-torso args)))) (setq use-torso nil))
     (cond
       ((null ctype)
	(send self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso use-torso :clear-velocities t))
       ((equal ctype :arm-controller)
	(send self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso nil :clear-velocities t :ctype ctype))
       (t
	(send self :angle-vector-raw av tm ctype)))
     ))
```
Testing a bit by sending *fetch* angle vector of #f(222.0 22.0 22.0 22.0 22.0 22.0 22.0 22.0 44.0 44.0) to *ri* set in :init-pose i.e. #f(0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0) got me the following result:
```
Evaluating: (send *ri* :angle-vector (send *fetch* :angle-vector) 3000) ...

[ INFO] [1493298111.846125819, 4928.413000000]: ;; Planned Trajectory Total Time   2.356 [sec]
[ INFO] [1493298111.846394308, 4928.413000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1493298111.846442748, 4928.413000000]: ;; generated 10 points for 3 sec using [arm_with_torso] group
	=> (#f(222.0 22.0 22.0 22.0 22.0 22.0 22.0 22.0 44.0 44.0))
	potentio-vector=> #f(222.369 22.0132 22.0385 22.2371 21.9771 22.1409 21.9988 22.1707 43.897 44.2347)

Evaluating: (send *ri* :angle-vector (send *fetch* :angle-vector) 3000 :use-torso nil) ...

[ INFO] [1493298121.077164733, 4935.738000000]: ;; Planned Trajectory Total Time   1.268 [sec]
[ INFO] [1493298121.077535501, 4935.738000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1493298121.077572813, 4935.738000000]: ;; generated 10 points for 3 sec using [arm] group
	=> (#f(222.0 22.0 22.0 22.0 22.0 22.0 22.0 22.0 44.0 44.0))
	potentio-vector=> #f(221.978 21.9968 22.0038 22.081 21.9961 22.1506 21.9752 22.042 43.9066 44.2285)

Evaluating: (send *ri* :angle-vector (send *fetch* :angle-vector) 3000 :arm-controller) ...

[ INFO] [1493298130.387027014, 4943.188000000]: ;; Planned Trajectory Total Time   1.271 [sec]
[ INFO] [1493298130.387269279, 4943.188000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1493298130.387302321, 4943.188000000]: ;; generated 10 points for 3 sec using [arm] group
	=> (#f(222.0 22.0 22.0 22.0 22.0 22.0 22.0 22.0 44.0 44.0))
	potentio-vector=> #f(0.867726 22.0219 22.0359 22.057 22.0054 22.1955 22.0046 22.0313 -0.032111 0.136118)

Evaluating: (send *ri* :angle-vector (send *fetch* :angle-vector) 3000 :torso-controller) ...

	=> #f(222.0 22.0 22.0 22.0 22.0 22.0 22.0 22.0 44.0 44.0)
	potentio-vector=> #f(221.983 0.007242 0.000782 -0.092961 0.040565 0.60528 -0.034643 0.02205 -0.068102 0.12147)

Evaluating: (send *ri* :angle-vector (send *fetch* :angle-vector) 3000 :head-controller) ...

	=> #f(222.0 22.0 22.0 22.0 22.0 22.0 22.0 22.0 44.0 44.0)
	potentio-vector=> #f(0.470645 -0.002691 -0.016818 -0.08118 -0.042216 0.519754 0.04909 -0.02716 43.9873 44.1763)
```

@knorth55 There was no problem in the IK that made the head move. The start of this was that in order to get a better point of view for rosbag Yamaguchi ( @708yamaguchi ) adjusted the head joint with the remote controller, which kept being corrected every time it sent angle-vector to robot-interface, and when I suggested using arm-controller I realized that it wasn't working for fetch.

@k-okada Using the name :controller-type got me the following error:
```
/opt/ros/indigo/share/euslisp/jskeus/eus/Linux64/bin/irteusgl 0 error: list exp\
ected in (mapcar #'(lambda (action param) (send self :send-trajectory-each acti\
on (cdr (assoc :joint-names param)) joint-trajectory-msg starttime)) ca (send s\
elf ct))
```
Using :ctype went fine to cases like `(send *ri* :angle-vector (send *fetch* :angle-vector) 3000 :arm-controller :use-torso nil)`, but not so good for `(send *ri* :angle-vector (send *fetch* :angle-vector) :use-torso nil)` and not rather what expected for everything else. Thus resulted in making the changes above.
Of course the ideal should be, as you mentioned, something like `(send *ri* :rarm :angle-vector (send *fetch* :rarm :angle-vector))`, which would also return something more meaningful than when using controllers.
About defining new methods and its procedure, I fully understand the importance and reliability of an API being used for over 20 years. However, the source code is far from perfect, and there is a lot of improvement that can be done. It seems to me that it could be a good idea to have something like a "development" branch for handling this kind of changes, but I do not know if JSK would be willing to do such a thing.
Since we are in this topic, I actually want to ask a lit bit about the common lisp part of euslisp too. I hear that euslisp was born as an extended language of common lisp, but nowadays there seems to be a relevant part of common lisp not compatible with euslisp. This would be completely ok if euslisp had detailed and sturdy documentation, but it is hard for me to say that it does. I do not know if JSK is interested in increasing the euslisp user community or not, but if it is it might be good to update the language built in functions or release some tutorials on it.